iModelCore/ECPresentation/Tests/NonPublished/Integration/Content/ContentIntegrationTests.cpp

```diff
@@ -5735,6 +5735,364 @@ TEST_F(RulesDrivenECPresentationManagerContentTests, ContentInstancesOfSpecificC
+
+    ASSERT_TRUE(descriptor.IsValid());
+    EXPECT_EQ(1, descriptor->GetVisibleFields().size()); // A -> B
+    EXPECT_EQ(classB->GetName(), descriptor->GetVisibleFields()[0]->GetLabel());
+    auto fieldsB = descriptor->GetVisibleFields()[0]->AsNestedContentField()->GetFields();
+    EXPECT_EQ(2, fieldsB.size()); // backward A -> B and PropertyB
+    EXPECT_STREQ("PropertyB", fieldsB[0]->GetLabel().c_str());
+    EXPECT_EQ(classA->GetName(), fieldsB[1]->GetLabel());
+    auto nestedFieldsA = fieldsB[1]->AsNestedContentField()->GetFields();
+    EXPECT_EQ(1, nestedFieldsA.size()); // A -> B
+    EXPECT_EQ(classB->GetName(), nestedFieldsA[0]->GetLabel());
+    auto nestedFieldsB = nestedFieldsA[0]->AsNestedContentField()->GetFields();
+    EXPECT_EQ(1, nestedFieldsB.size()); // PropertyB
+    EXPECT_STREQ("PropertyB", nestedFieldsB[0]->GetLabel().c_str());
+
+    RapidJsonValueCR propertyB = jsonValues[NESTED_CONTENT_FIELD_NAME(classA, classB)][0]["Values"];
+    EXPECT_STREQ("InstanceB", propertyB[FIELD_NAME(classB, "PropertyB")].GetString());
+    RapidJsonValueCR nestedPropertyA = propertyB[NESTED_CONTENT_FIELD_NAME(classB, classA)][0]["Values"];
+    EXPECT_STREQ("InstanceB", nestedPropertyA[NESTED_CONTENT_FIELD_NAME_C(classA, classB, 2)][0]["Values"][FIELD_NAME_C(classB, "PropertyB", 2)].GetString());
+    }
+
+/*---------------------------------------------------------------------------------**//**
+* @bsitest
++---------------+---------------+---------------+---------------+---------------+------*/
+DEFINE_SCHEMA(ContentInstancesOfSpecificClassesSpecification_ContentModifierShouldNotAddRelatedPropertiesOnNestedContentWhenNotAllPropertiesIncluded, R"*(
+    <ECEntityClass typeName="A">
+    </ECEntityClass>
+    <ECEntityClass typeName="B">
+        <ECProperty propertyName="PropertyB" typeName="string" />
+    </ECEntityClass>
+    <ECEntityClass typeName="C">
+    </ECEntityClass>
+    <ECRelationshipClass typeName="A_Has_B" strength="referencing" strengthDirection="forward" modifier="None">
+        <Source multiplicity="(0..1)" roleLabel="references" polymorphic="True">
+            <Class class="A" />
+        </Source>
+        <Target multiplicity="(0..1)" roleLabel="is referenced by" polymorphic="True">
+            <Class class="B" />
+        </Target>
+    </ECRelationshipClass>
+    <ECRelationshipClass typeName="B_Has_C" strength="referencing" strengthDirection="forward" modifier="None">
+        <Source multiplicity="(0..1)" roleLabel="references" polymorphic="True">
+            <Class class="B" />
+        </Source>
+        <Target multiplicity="(0..1)" roleLabel="is referenced by" polymorphic="True">
+            <Class class="C" />
+        </Target>
+    </ECRelationshipClass>
+)*");
+TEST_F(RulesDrivenECPresentationManagerContentTests, ContentInstancesOfSpecificClassesSpecification_ContentModifierShouldNotAddRelatedPropertiesOnNestedContentWhenNotAllPropertiesIncluded)

```

Comment:
I'd like a similar test for calculated properties

Reply:
made some changes when adding calculated properties. Please check